### PR TITLE
Apache Spark Notebook

### PR DESCRIPTION
**Pull request by anabranch**
Fixes the contributing & publishing problem featured in issue #21

### CODE REVIEW

**Comment by anabranch** [comment]
sorry it includes my commits from the previous notebook

File: _makescripts/publish.py [hunk 1]
```diff
@@ -37,7 +37,8 @@ def img_alt(img_src, img_i):
 # Remove title
 def remove_title(body):
     H1 = body.findAll('h1')  # TODO check 'h2' also?
-    H1[0].extract()
+    if H1:
```
Comment (etpinard) [inline _makescripts/publish.py]: Nice catch. Thank you very much.

File: notebooks/references.json [hunk 2]
```diff
@@ -8,6 +8,7 @@
         "networkx",
         "make_subplots",
         "basemap",
-        "collaborate"
+        "collaborate",
+        "apachespark"
```
Comment (etpinard) [inline notebooks/references.json]: `"apachespark"` should be at the top of the list.

File: notebooks/apachespark/config.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+    "title": "Plotting Spark DataFrames and Plotly",
+    "title_short": "Apache PySpark",
+    "meta_description": "A tutorial showing how to plot Apache Spark DataFrames with Plotly",
+    "cells": [0, -1],
+    "relative_url": "Apache-Spark",
```
Comment (etpinard) [inline notebooks/apachespark/config.json]: This should be 

- `"cells": [0, "end"]` as we want to include all cells in the published version.
- `"relative_url": "apache-spark"` as lower case looks better.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -109,6 +109,8 @@ and fill in the generated notebook `config.json`
 
 #### Step 3:
 
+Add your notebook name into `references.json` in the notebooks sub-directory then run:
```
Comment (etpinard) [inline CONTRIBUTING.md]: thanks

File: notebooks/apachespark/config.json (hunk 3 above)
Comment (anabranch) [inline notebooks/apachespark/config.json]: I think this should be the default then, I figured -1 meant all the cells.

**Comment by etpinard** [comment]
@anabranch Almost there.

- Two small edits to the NB `config.json`
- and a reordering in `references.json`


then `make run-publish nb=apachespark` is all there's left.

File: notebooks/apachespark/config.json (hunk 3 above)
Comment (etpinard) [inline notebooks/apachespark/config.json]: Thanks for the suggestion.

Most NBs in this repo have a CSS cell like

image

which we remove from the published version; hence, the `"cells: [0, -1]` default.

As more NBs are added, maybe a `[0, "end"]` default would be better. Time will tell.

By the way, the `cells` field behaviour is documented [here](https://github.com/plotly/IPython-plotly/blob/master/CONTRIBUTING.md#step-2)

**Comment by anabranch** [comment]
@etpinard 
Think we should be OK, I closed the other branch because it seems like you're ok just committing it all here. Let me know if I need to separate them again.

**Comment by etpinard** [comment]
Unfortunately https://github.com/plotly/IPython-plotly/pull/19 got merged before your PR and is causing some conflicts.

You can either try to resolve them yourself or I can take it from here if you want.. Your call.

**Comment by anabranch** [comment]
@etpinard should be good!

**Comment by anabranch** [comment]
@etpinard Was looking to reference this online and saw that this hasn't been merged yet. Any idea on when that'll go through?

**Comment by anabranch** [comment]
@etpinard just went ahead and fixed it for you!

**Comment by etpinard** [comment]
Beautiful. Thanks again.

**Comment by etpinard** [comment]
@anabranch -> https://plot.ly/ipython-notebooks/apache-spark/

**Pull request merged**